core/src/main/java/org/apache/brooklyn/location/byon/FixedListMachineProvisioningLocation.java

```diff
@@ -345,7 +346,7 @@ protected void updateMachineConfig(T machine, Map<?, ?> flags) {
             // For backwards compatibility, where peristed state did not have this.
             origConfigs = Maps.newLinkedHashMap();
         }
-        Map<String, Object> strFlags = ConfigBag.newInstance(flags).getAllConfig();
+        Map<String, Object> strFlags = ConfigBag.newInstanceExtending(((AbstractConfigurationSupportInternal)machine.config()).getBag(), flags).getAllConfig();
```

Comment:
Does this need to be applied to the other implementations of `MachineProvisioningLocation.obtain()`?

Will this effect the behaviour of existing blueprints?